### PR TITLE
Update Azure Service Bus components to track2 SDK

### LINKED ISSUE #1531: [Azure][Track 2 SDK] Migrate Azure Service Bus Queues Binding
Migrate Azure Service Bus Queues Binding to SDK github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus

### LINKED ISSUE #1532: [Azure][Track 2 SDK] Migrate Azure Service Bus PubSub
Migrate Azure Service Bus PubSub to SDK github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
# Description

Move the Azure Service Bus components (binding and pubsub) to the new "track2" SDKs. Thanks a lot to @halspang for the initial work in #1648!

Thanks to the update and some refactoring work, two metadata keys for Azure Service Bus Pub/Sub are not needed anymore:

- `prefetchCount`
- `maxActiveMessagesRecoveryInSec`

Notes:

1. Because the Service Bus SDKs depended on the newer version of azcore and azidentity, I had to update the other Azure SDKs that were already in "track2". Some of the updates contained breaking changes in the APIs, which I've fixed.
1. ~~This requires Go 1.18 so it depends on #1697 (which depends on #1685 itself)~~

## Issue reference

Please reference the issue this PR will:
closes #1532 
closes #1531
closes #1701

### CODE REVIEW

File: pubsub/azure/servicebus/servicebus.go
Comment (ItalyPaleAle) [inline pubsub/azure/servicebus/servicebus.go:489]: This is unchanged, but wondering if we should make the app crash here (`Fatalf`)? Otherwise, the error would get logged and the app continues to work but without a connection to Service Bus.

File: pubsub/azure/servicebus/message_test.go
Comment (ItalyPaleAle) [inline pubsub/azure/servicebus/message_test.go:141]: This test can't be implemented with track2 SDKs. The method to create these objects is un-exported.

Can be re-added if https://github.com/Azure/azure-sdk-for-go/pull/17848 is merged

**Comment by berndverst** [comment]
@ItalyPaleAle please fix linter issues. Also use `/ok-to-test` to trigger actions for conformance and certification tests (separately from this PR) to run the Service Bus Queues binding certification test.

**Comment by berndverst** [comment]
At the moment this breaks EventHubs Binding, PubSub and Service Bus Queues Binding.

Please take a look by running the certification and conformance tests via `/ok-to-test`

**Comment by ItalyPaleAle** [comment]
@berndverst This is ready for review IMHO.

- All tests are passing (waiting for the conformance tests to complete)
- I have confirmed that both the binding and pubsub are able to recover successfully in case of failures (both pubsub and input/output bindings)

Once this is merged, I'll need to update the docs because of metadata changes, as well as our E2E tests if needed.
-

**Comment by berndverst** [comment]
Eventhubs Binding conformance test is failing
Eventhubs Binding certification test is failing
Service Bus Queues Binding certification test is failing

(you can see in the results of your /ok-to-test run)

**Comment by ItalyPaleAle** [comment]
I'll look into the Service Bus binding failure tomorrow.

For Event Hubs, that's very odd because I did not touch that module besides for a minimal change that was required. However, it looked like the SDK got upgraded (which was un-intentional). I've reverted the change to the SDK and let's see if that fixes the tests.

**Comment by ItalyPaleAle** [comment]
@berndverst  I have fixed all tests that were failing because of changes in my code.

After fixing dependencies, there were 2 tests that fail.

- Service Bus Queue binding: when a message fails to be processed by a handler, it's inserted back into the queue. This means that it's re-delivered later, so it may not come in order. However the conf tests were expecting messages to be in order. I think this was a problem with the tests as the re-enqueueing of the message is arguably the desired behavior
- Event Hubs binding: this is still failing and it looks like it was failing before 100% of the times when running on master. See: https://github.com/dapr/components-contrib/actions/workflows/certification.yml The error in this PR is the same. The test is broken and needs to be fixed, but it should not block this PR.

(On why the Event Hubs test is failing: the test `TestAzureServiceBusQueueRetriesOnError` is creating 2 Event Hubs listeners with the same consumer group, which seems to be disallowed)

**Review by berndverst — APPROVED** [review]

**Pull request merged**
